preview/preview.js

```diff
@@ -0,0 +1,87 @@
+const child_process = require('child_process');
+const http = require('http');
+const url = require('url');
+
+const port = 3000;
+const checkTypeOpts = {
+  'cwd': '/docs_build/.repos/target_repo.git',
+  'max_buffer': 64,
+};
+const catOpts = {
+  'cwd': '/docs_build/.repos/target_repo.git',
+  'encoding': 'buffer',
+  'max_buffer': 1024 * 1024 * 20,
+};
+
+const requestHandler = (request, response) => {
+  parsedUrl = url.parse(request.url);
+  if (false == parsedUrl.pathname.startsWith('/guide')) {
```

Comment:
nit: `!parsedUrl`. Also, always default to strict equals aka `===` in JS, as `==` is a bit looser e.g. `10 == '10'` => true